src/app/containers/MediaPlayer/helpers/embedUrl.js

```diff
@@ -1,13 +1,29 @@
-const embedUrl = ({ type, requestUrl, isAmp = false }) => {
-  const avRoute = 'ws/av-embeds';
-  const baseUrl = process.env.SIMORGH_EMBEDS_BASE_URL;
-  const parts = [baseUrl, avRoute, type, requestUrl];
+import isLive from '#lib/utilities/envHelper';
+import hasLiveOverride from '#lib/utilities/rendererEnvHelper';
+import nodeLogger from '#lib/logger.node';
+
+const logger = nodeLogger(__filename);
+
+const AV_ROUTE = 'ws/av-embeds';
+
+const getBaseUrl = url => {
+  if (hasLiveOverride(url) || isLive()) {
+    return process.env.SIMORGH_EMBEDS_BASE_URL_LIVE;
+  }
+  return process.env.SIMORGH_EMBEDS_BASE_URL_TEST;
+};
+
+const embedUrl = ({ type, mediaId, isAmp = false, pageUrl }) => {
+  const urlParts = [getBaseUrl(pageUrl), AV_ROUTE, type, mediaId];
 
   if (isAmp) {
-    parts.push('amp');
+    urlParts.push('amp');
   }
 
-  return parts.join('/');
+  const urlEmbed = urlParts.join('/');
+  logger.info(`Embed URL: ${urlEmbed}`);
+
+  return urlEmbed;
```

Comment:
instead of working with arrays and mutating with .push why not try something like this using string interpolation

```js
const baseUrl = getBaseUrl(pageUrl)
const url = `${baseUrl}/${AV_ROUTE}/${type}/${mediaId}`

return isAmp ? `${url}/amp` : url
```

Reply:
Nice, I'll do that!

Reply:
Nice, I'll do that